internal/services/compute/proximity_placement_group_resource.go

```diff
@@ -47,6 +50,24 @@ func resourceProximityPlacementGroup() *pluginsdk.Resource {
 
 			"location": commonschema.Location(),
 
+			"intent_vm_sizes": {
+				Type:     pluginsdk.TypeSet,
+				Optional: true,
+				MinItems: 1,
+				Elem: &pluginsdk.Schema{
+					Type:         pluginsdk.TypeString,
+					ValidateFunc: validation.StringIsNotWhiteSpace,
+				},
+			},
+
+			"zone": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				RequiredWith: []string{"intent_vm_sizes"},
+				ForceNew:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
```

Comment:
we should be using `commonschema.ZoneSingle` here

Reply:
`zone` in this resource is set to `RequiredWith: []string{"intent_vm_sizes"}`, thus using a separate schema.